demo/src/settings/preferences-provider.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Utility class for loading and saving preferences to local storage.
+ *
+ * @class PreferencesProvider
+ */
+class PreferencesProvider {
+  /**
+   * Load preferences from local storage.
+   *
+   * @static
+   * @returns {Object} An object representing the loaded preferences.
+   */
+  static loadPreferences() {
+    return JSON.parse(localStorage.getItem('preferences')) || {};
```

Comment:
This may fail, as there are cases where `localStorage` can be `undefined`. But we can leave it like that for now